app/Infrastructure/DoctrineUserRepository.php

```diff
@@ -24,4 +23,9 @@ public function insert(User $user): int
         $this->em->flush();
         return $user->id;
     }
+
+    public function findBy(string $property, string $valueProperty): User | null
+    {
+        return $this->em->getRepository(User::class)->findOneBy(array($property => $valueProperty));
```

Comment:
```suggestion
        return $this->em->getRepository(User::class)->findOneBy(array($property => $valueProperty));
```

O que significa `em` ? `entity mapping` ?
Sugestão, evitar abreviações ( clean code ).

Reply:
O `em` significa `EntityManager`, vi alguns projetos que usam doctrine e é bem comum eles utilizarem `em` para fazer nomear, por isso segui esse padrão

Reply:
seria o caso de ser mais verboso e usar $this->entityManager, ou é merda demais ? :smiley:

Reply:
Sim, poderia chamar de entityManager tb sem problemas, é que acho que a galera que trabalha com Doctrine sabe que sempre precisa do Entity Manager então é meio que padrão usar o em

Reply:
Galera, sobre esse tópico, vou manter como `em` visto que é fortemente adotado pela comunidade de PHP esta nomenclatura quando se refere ao `EntityManager` do doctrine 

Na própria [documentação](https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/reference/working-with-objects.html#persisting-entities) do Doctrine eles utilizam desse padrão